Intersect.Client/Entities/Player.cs

```diff
@@ -2252,124 +2252,71 @@ public virtual void DrawGuildName(Color textColor, Color borderColor = null, Col
         }
 
         // Draw Entities Overhead Information when hovering the cursor by them
-        // (when they are hidden by the game settings preferences).
+        // (when they are hidden by game settings preferences).
         public void DrawOverheadInfoOnHover()
         {
-            if (Interface.Interface.MouseHitGui())
+            var map = this.MapInstance;
+            if (map == null || Interface.Interface.MouseHitGui())
             {
                 return;
             }
 
-            var mousePos = Graphics.ConvertToWorldPoint(Globals.InputManager.GetMousePosition());
-            foreach (MapInstance map in Maps.MapInstance.Lookup.Values)
```

Comment:
This looks functionally different from what you changed it to.

Reply:
we already exited early if this.MapInstance is null, we are also exiting early if the entities map id doesn't belongs to that map instance, thus, we don't really need all these checks and loops anymore:

```
foreach (MapInstance map in Maps.MapInstance.Lookup.Values)
            {
                if (mousePos.X >= map.GetX() && mousePos.X <= map.GetX() + Options.MapWidth * Options.TileWidth)
                {
                    if (mousePos.Y >= map.GetY() && mousePos.Y <= map.GetY() + Options.MapHeight * Options.TileHeight)
                    {
```

but instead, with less nested code and much more simple and less resource consuming, we do the following:

`en.Value.WorldPos.Contains(mousePos.X, mousePos.Y)`

from which we have already filtered out the entities we don't want to check after.

Reply:
But the previous loop checked to see if an entity belonged to the map instance that contains the world coordinate the mouse points to -- that is not necessarily `this.MapInstance`.

If there is an syntax change that needs to happen it would be to refactor the bounds checks into a method and then invert the checks into `continue;` statements to skip additional maps.

If there is a performance optimization that needs to happen, I would think it would be just looping through the entities of the given map, not looping through the global entities (why are we even doing this?).

Reply:
excluding the entities from other maps lead to the issue where we can't hover/see entities info outside the current map.

Now, how about !en.Value.MapInstance.InView() ? 

![imagen](https://user-images.githubusercontent.com/17498701/210473605-873d639e-fefd-47d0-8994-22cfda8a1a37.png)

![imagen](https://user-images.githubusercontent.com/17498701/210473619-74b9c2b7-5b04-4929-8daa-ccc43ae9ea77.png)

seems to behave as intended but i'm not sure if it's really allowing me to exclude computing within the loop all of the entities

Reply:
> If there is a performance optimization that needs to happen, I would think it would be just looping through the entities of the given map, not looping through the global entities (why are we even doing this?).

This approach is possible with the drawback i mentioned in the previous comment (inability to hover over entities outside the current map boundaries - which i _personally_ don't mind since i use unconnected maps, but it may be annoying to users doing so)